File: 品味Java的21種設計模式/dp_java/visitor/example3/Element.java
package cn.javass.dp.visitor.example3;
/**
 * 被访问的元素的接口
 */
public abstract class Element {
    /**
     * 接受访问者的访问
     * @param visitor 访问者对象
     */
    public abstract void accept(Visitor visitor);
}

File: 23種設計模式及對應C++示例代碼/行为模式/Visitor/Visitor_exam/Element.cpp
// Element.cpp: implementation of the Element class.
//
//////////////////////////////////////////////////////////////////////

#include "stdafx.h"
#include "Element.h"
#include "Visitor.h"


//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////


void ConcreteElementA::Accept(Visitor* vis)
{
	cout<<"即将调用元素A的方法，";
	vis->VisitConcreteElementA(this);	
}
void ConcreteElementB::Accept(Visitor* vis)
{
	cout<<"即将调用元素B的方法，";
	vis->VisitConcreteElementB(this);
}
